### ISSUE TITLE
Modulo parenthesization

### ISSUE THREAD

**Issue by bjodah** [sympy/sympy#23912]
First off, I'm not sure if there's a universally agreed upon definition of what the modulo operator should return for complex numbers, python's built in operator raises a `TypeError`.

Regardless of that, the parenthesization currently looks dubious and should perhaps be fixed?
```python
>>> Mod(3+2j,4+5j)
(3.0 + 2.0⋅ⅈ) mod 4.0 + 5.0⋅ⅈ
```

**Comment by oscarbenjamin** [comment]
> modulo operator should return for complex numbers

I'm pretty sure there's an old SymPy issue somewhere that discusses this at length.

**Comment by bjodah** [comment]
You're correct, not sure how I missed this issue during my first survey of old issues: gh-11391
The printing / operator precedence issue still stands I guess.

**Comment by Jay-Patankar** [comment]
@bjodah Is this issue still open for solving?

**Comment by asmeurer** [comment]
Just ignoring complex numbers it doesn't handle Add correctly.

```
>>> pprint(Mod(x, y + z))
x mod y + z
```

**Comment by demohatesrng** [comment]
it can indeed cause confusion cz of its current representation to how it is. sympy's mod function isn't intended for complex numbers i think it is only valid for integers . Someone pls correct me if i am wrong
